src/compiler/checker.ts

```diff
@@ -22760,6 +22762,19 @@ namespace ts {
                     return Debug.fail();
             }
 
+            function checkTypeComparableWithLessOrGreaterThanOperator(valueType: Type): boolean {
+                const t = valueType.flags;
+                return (t === TypeFlags.String) ||

```

Comment:
I'm pretty sure all these comparisons should be using `&` and not `===`. I'm also curious why `NumberLiteral` is missing - I'd assume comparing `0` and `2` is probably OK.

Reply:
I'd definitely like to see some tests comparing numeric literal types of all sorts among one another and with their bases (`number`, `bigint`).

Reply:
I have lots of troubles adapting the tests. I prefer to give up on this PR :(